cmd/ovirt-populator/ovirt-populator.go

```diff
@@ -34,171 +34,157 @@ type TransferProgress struct {
+		},
+		[]string{"ownerUID"},
+	)
+	if err := prometheus.Register(progressCounter); err != nil {
+		klog.Error("Prometheus progress gauge not registered:", err)
+		return
 	} else {
-		// for secure connection use the ca cert
-		cert, err := os.Create("/tmp/ca.pem")
-		if err != nil {
-			klog.Fatalf("Failed to create ca.pem %v", err)
-		}
+		klog.Info("Prometheus progress gauge registered.")
+	}
 
-		defer cert.Close()
-		_, err = cert.Write([]byte(config.cacert))
-		if err != nil {
-			klog.Fatalf("Failed to write CA to file: %v", err)
+	var currentProgress float64
+	total := pvcSize
+	metric := &dto.Metric{}
+
+	for scanner.Scan() {
+		progressOutput := TransferProgress{}
+		text := scanner.Text()
+		klog.Info(text)
+		if err := json.Unmarshal([]byte(text), &progressOutput); err != nil {
+			var syntaxError *json.SyntaxError
+			if !errors.As(err, &syntaxError) {
+				klog.Error(err)
+			}
 		}
-
-		return []string{
-			"download-disk",
-			"--output", "json",
-			"--engine-url=" + config.URL,
-			"--username=" + config.username,
-			"--password-file=/tmp/ovirt.pass",
-			"--cafile=" + "/tmp/ca.pem",
-			"-f", "raw",
-			diskID,
-			volPath,
+		if total > 0 {
+			currentProgress = (float64(progressOutput.Transferred) / float64(total)) * 100
+			if err := progressCounter.WithLabelValues(ownerUID).Write(metric); err != nil {
+				klog.Error(err)
+			} else if currentProgress > metric.Counter.GetValue() {
+				progressCounter.WithLabelValues(ownerUID).Add(currentProgress - metric.Counter.GetValue())
+			}
 		}
 	}
-}
-
-func loadEngineConfig(engineURL string) engineConfig {
-	user := os.Getenv("user")
-	pass := os.Getenv("password")
 
-	insecureSkipVerify, found := os.LookupEnv("insecureSkipVerify")
-	if !found {
-		insecureSkipVerify = "false"
+	metric.GetCounter().Value = ptr.To[float64](100)
+	if err := progressCounter.WithLabelValues(ownerUID).Write(metric); err != nil {
+		klog.Error(err)
 	}
 
-	insecure, err := strconv.ParseBool(string(insecureSkipVerify))
-	if err != nil {
-		klog.Fatal(err.Error())
+	done <- struct{}{}
+}
+
+func createCommandArguments(config *engineConfig, diskID, volPath string) []string {
+	var args []string
+	args = append(args, "download-disk", "--output", "json", "--engine-url="+config.URL, "--username="+config.username, "--password-file=/tmp/ovirt.pass")
+
+	if config.insecure {
+		args = append(args, "--insecure")
+	} else {
+		args = append(args, "--cafile=/tmp/ca.pem")
 	}
 
-	//If the insecure option is set, the ca file field in the secret is not required.
-	cacert := os.Getenv("cacert")
+	args = append(args, "-f", "raw", diskID, volPath)
+	return args
+}
 
-	return engineConfig{
+func loadEngineConfig(engineURL string) *engineConfig {
+	user, pass := os.Getenv("user"), os.Getenv("password")
+	insecure := getEnvAsBool("insecureSkipVerify", false)
+	return &engineConfig{
 		URL:      engineURL,
 		username: user,
 		password: pass,
-		cacert:   cacert,
+		cacert:   os.Getenv("cacert"),
```

Comment:
what is the reason to change it here? and if so, why leaving the rest of the `GetEnv` above?

Reply:
changed